### PR TITLE
Implement UI setting to hide listed Position Attributes

### LINKED ISSUE #578: Feature Request: possible to deactivate by displaying flags in the status panel
Hello
Anton would it be possible to deactivate by displaying flags in the status panel some fields that are not necessary for the final customer?
Example:
Operator, Hdop, Pdop, IP, etc

### PR DESCRIPTION
**Pull request by Abyss777**
Implemented user/server attribute to set list of hidden Position Attributes

image

fix #578

### CODE REVIEW

File: web/app/view/StateController.js [hunk 1]
```diff
@@ -50,11 +50,20 @@ Ext.define('Traccar.view.StateController', {
...
+        var i, hiddenAttributesPreference, attributesList;
         if (Traccar.app.getUser().get('admin') ||
                 !Traccar.app.getUser().get('deviceReadonly') && !Traccar.app.getPreference('readonly', false)) {
             this.lookupReference('computedAttributesButton').setDisabled(
                 Traccar.app.getBooleanAttributePreference('ui.disableComputedAttributes'));
         }
+        hiddenAttributesPreference = Traccar.app.getAttributePreference('ui.hiddenPositionAttributes');
+        this.hiddenAttributes = {};
+        if (hiddenAttributesPreference) {
+            attributesList = hiddenAttributesPreference.split(/[ ,]+/).filter(Boolean);
```
Comment (Abyss777) [inline web/app/view/StateController.js]: List can be comma or space separated (or both). Empty values are filtered.

File: web/app/view/StateController.js [hunk 2]
```diff
@@ -50,11 +50,20 @@ Ext.define('Traccar.view.StateController', {
...
+            for (i = 0; i < attributesList.length; i++) {
+                this.hiddenAttributes[attributesList[i]] = true;
```
Comment (Abyss777) [inline web/app/view/StateController.js]: Initialize list once on init for speed.

File: web/app/store/CommonUserAttributes.js [hunk 3]
```diff
@@ -59,6 +59,10 @@ Ext.define('Traccar.store.CommonUserAttributes', {
         name: Strings.attributeUiDisableCalendars,
         valueType: 'boolean'
     }, {
+        key: 'ui.hiddenPositionAttributes',
```
Comment (tananaev) [inline web/app/store/CommonUserAttributes.js]: I think it should be called `hide` instead of `hidden`.

**Comment by Abyss777** [comment]
Renamed the parameter and related strings

File: web/app/view/StateController.js [hunk 4]
```diff
@@ -50,11 +50,20 @@ Ext.define('Traccar.view.StateController', {
 
 
     init: function () {
+        var i, hiddenAttributesPreference, attributesList;
```
Comment (tananaev) [inline web/app/view/StateController.js]: There are still `hidden` in some places.

**Comment by Abyss777** [comment]
Oh, sorry, replaced the commit.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**